src/docs/arch-index.dox

```diff
@@ -121,15 +123,75 @@ wt_file -[hidden]right-> log_file
 
 We go into some detail for some of the internal components.
 
+@subpage arch-block
+
+The Block Manager manages the reading and writing of disk blocks.
+
+@subpage arch-cache
+
+Cache is represented by the various shared data structures that
+make up in-memory Btrees and subordinate data structures.
+
+@subpage arch-column
+
+Column Stores are Btrees that have as their key a record id.
+
+@subpage arch-cursor
+
+Cursors are used to get and modify data.
+
+@subpage arch-data-file
+
+The format of the data file is given by structures in \c block.h .
+
 @subpage arch-dhandle
 
 An internal structure called a Data Handle (dhandle) is used to represent and
 access Btrees and other data sources in WiredTiger.
 
+@subpage arch-eviction
+
+Eviction represents the process of writing dirty data to disk.
```

Comment:
See note above.  Maybe, "Eviction represents the process or removing old data from the cache, writing it to disk if it is dirty"?

Reply:
I reworked this slightly, starting with your words.